### PR TITLE
ApplyColorMap can be used with a user colormap

### PR DESCRIPTION
**Pull request by LaurentBerger**
### This pullrequest changes

applyColorMap can be used now with a user colormap defined in a Mat. In LUT function when image is CV_8UC1 lut table must be CV_8UC1 too. In applyColorMap it can be CV_8UC1 in source image with a color table  of type CV_8UC3

### CODE REVIEW

**Comment by alalek** [comment]
Please add a test for new functionality.

**Comment by mshabunin** [comment]
@LaurentBerger , why do you need to apply CV_8UC3 color map to CV_8UC1 image?

**Comment by alalek** [comment]
> CV_8UC3 and size 256

This is something strange. Do you have use case for this?

**Comment by LaurentBerger** [comment]
@mshabunin When I have infrared image or connectedComponents (keeeping only lowsignificant bits)  I need it

**Comment by alalek** [comment]
BTW, Looks like the full implementation of proposed function is (without parameters check):
```
if(src.type() == CV_8UC3)
    cvtColor(src.clone(), src, COLOR_BGR2GRAY);
cvtColor(src.clone(), src, COLOR_GRAY2BGR);
LUT(src, userColor, dst);
```
So it is just a LUT call with 1-2 cvtColor.

**Comment by LaurentBerger** [comment]
@alalek About 

> CV_8UC3 and size 256
I will check it if PR is useful

> So it is just a LUT call with 1-2 cvtColor.
Yes I'm agree. My PR is only apply a color map to a Mat to display Mat data

File: modules/imgproc/src/colormap.cpp [hunk 1]
```diff
@@ -546,4 +562,21 @@ namespace colormap
...
+        if (userColor.type() != CV_8UC1 && userColor.type() != CV_8UC3)
+            CV_Error(Error::StsAssert, "cv::LUT only supports tables CV_8UC1 or CV_8UC3.");
+        colormap::ColorMap* cm = (colormap::ColorMap*) (new colormap::UserColorMap(userColor.getMat()));
+
+        if (!cm)
+            CV_Error(Error::StsBadArg, "Unknown inputArray");
+
+        (*cm)(src, dst);
+
+        delete cm;
```
Comment (alalek) [inline modules/imgproc/src/colormap.cpp]: Why new/delete are required here? (I see here local usage of cm variable):
```
colormap::UserColorMap cm(userColor.getMat());
cm(src, dst);
```

**Comment by vpisarev** [comment]
let's put it in. It does not add any extra loop, it's just different entry to the existing functionality :+1:

**Pull request merged**